base/number.jl

```diff
@@ -314,7 +314,7 @@ julia> oneunit(3.7)
 1.0
 
 julia> import Dates; oneunit(Dates.Day)
-1 day
+Dates.Day(1)
```

Comment:
This isn't quite right --- `1 day` should still be printed in the REPL, but `repr` should give `Dates.Day(1)`. The `BitPeriod` type can be removed, and instead the current `show` method should be a method `show(io::IO, ::MIME"text/plain", p::Period)`, and your new method might not be necessary since the fallback 2-argument show will probably print `Dates.Day(1)`.

Reply:
Right, I see what you're saying. I think my latest change should satisfy those requirements, while also satisfying `1 day` being shown in outputs such as in a dataframe.